stdlib/Mmap/src/Mmap.jl

```diff
@@ -212,6 +212,8 @@ function mmap(io::IO,
     @static if Sys.isunix()
         prot, flags, iswrite = settings(file_desc, shared)
         iswrite && grow && grow!(io, offset, len)
+        szfile = convert(Csize_t, len + offset)
+        !iswrite && szfile > filesize(io) && throw(ArgumentError("unable to increase file size to $szfile due to read-only permissions"))
```

Comment:
```suggestion
        szfile = convert(Csize_t, len + offset)
        if iswrite && grow
            grow!(io, offset, len)
        elseif szfile > filesize(io)
            throw(ArgumentError("unable to increase file size to $szfile due to read-only permissions"))
        end
```

Can you make this same change to the Windows code (to respect the `grow` parameter):
```julia
        (readonly || !grow) && szfile > filesize(io) && throw(ArgumentError("unable to increase file size to $szfile due to read-only permissions"))
```